src/routes/user.route.js

```diff
@@ -0,0 +1,30 @@
+import { Router } from 'express';
+import jwt from 'jsonwebtoken';
+import googlePassport from '../controllers/users/user';
+
+const userRouter = new Router();
+
+userRouter.use(googlePassport.initialize());
+
+userRouter.get(
+  '/auth/google',
+  googlePassport.authenticate('google', { scope: ['profile', 'email'] })
+);
+
+userRouter.get(
+  '/auth/google/callback',
+  googlePassport.authenticate('google', { failureRedirect: '/login' }),
+  (req, res) => {
+    try {
+      const secretKey = process.env.SECRET_KEY;
+      const token = jwt.sign({ _id: req.user._id }, secretKey, {
+        algorithm: 'HS256'
+      });
+      res.header('authentication', token).json({ token, user: req.user });
```

Comment:
The token should be returned as a part of body instead of the header. Also remember to add the status code, for instance, ```res.status(200).json({})```